### PR TITLE
chore: Moves platform-specific signal code out of `main`

### LINKED ISSUE #1383: Move platform-specific signal handling code out of `main.rs`
After Windows support was added, the `main.rs` started to contain a lot of duplicated code for handling interruption signals on `cfg(unix)` and `cfg(windows)`. We need to factor it out by putting it to a separate source file or even a new crate in `lib`. See https://github.com/timberio/vector/pull/1219#discussion_r354382434 for details.

### PR DESCRIPTION
**Pull request by ktff**
Closes #1383 

With moving signal code "slight" updates to usage of `futures01` in handling code was applied. We can split this update out to a separate PR if needed, it was too tempting not to collapse all of the `select`,`compat`,`match`, etc., into a simpler form.

### CODE REVIEW

**Comment by lukesteensen** [comment]
Does need a `cargo fmt` before merge though.

**Comment by ktff** [comment]
@lukesteensen I'll be merging this, `cargo fmt` changed nothing, what did you see?

File: src/main.rs [hunk 1]
```diff
@@ -128,120 +132,67 @@ fn main() {
...
+                            {
+                                Ok(true) => (),
+                                Ok(false) => error!("Reload was not successful."),
+                                // Trigger graceful shutdown for what remains of the topology
+                                Err(()) => break SignalTo::Shutdown,
+                            }
+                        } else {
+                            error!("Reload aborted.");
+                        }
+                    }else{
```
Comment (lukesteensen) [inline src/main.rs:169]: I think this is all I saw. I wonder if the maco confuses rustfmt?

Comment (ktff) [reply src/main.rs:169]: It does, outside of the macro it formats it fine, I'll change it in other PR.

**Pull request merged**